### PR TITLE
OLED Support for MARIO 

### PR DESCRIPTION
**Pull request by ChinmayLonkar**
Hey @gautam-dev-maker 
I've added the OLED support for the MARIO bot, OLED will now display the MARIO 2k22 logo, and then after initialization and enabling, it will display the angles of the servo joints of the robot.

Kindly review this,

Regards,
Chinmay

### CODE REVIEW

File: examples/oled/main/main.c [hunk 1]
```diff
@@ -33,7 +33,7 @@ void app_main()
 	u8g2_t oled_config;
 
 	// Initialising the OLED
-	init_oled(&oled_config);
+	init_oled_walle(&oled_config);
```
Comment (gautam-dev-maker) [inline examples/oled/main/main.c:36]: sra-board-component is independent of WallE or MARIO, so why change the name of the function??

File: include/oled.h [hunk 2]
```diff
@@ -47,7 +47,16 @@
...
+esp_err_t init_oled_walle(u8g2_t *u8g2);
+
+/**
+ * @brief enables and configures OLED display
+
+ * @param u8g2 pointer to empty u8g2_t struct
+
+ * @return esp_err_t i.e it shows if OLED is initialised successfully or not
+ **/
+esp_err_t init_oled_mario(u8g2_t *u8g2);
```
Comment (gautam-dev-maker) [inline include/oled.h:59]: Why need for 2 initialiser function, even though we are using same oled model in both the workshop.

**Review by gautam-dev-maker — COMMENTED** [review]
How does MARIO logo looks on OLED, send a pic here.

**Comment by gautam-dev-maker** [comment]
Also this PR seems to have way too many changes, so for any further changes , consider creating new PR.

File: examples/oled/main/main.c (hunk 1 above)
Comment (ChinmayLonkar) [reply examples/oled/main/main.c:36]: The init_oled() function initiates the communication and displays SRA logo and the Walle Logo in the previous code.
But now I've made two functions separate for showing Walle and Mario logos in the initiation.

File: include/oled.h (hunk 2 above)
Comment (ChinmayLonkar) [reply include/oled.h:59]: the initializer function involves displaying Walle logo initially, so to display mario logo I've made a seperate initializer.
The solution can also be to not display the workshop logo in the initialization, instead we can do that after initialization.

**Comment by ChinmayLonkar** [comment]
> How does MARIO logo looks on OLED, send a pic here.

image

image

**Pull request closed without merging**